### PR TITLE
[NU-1301] Namespaced consumer groups and ObjectNaming refactor

### PR DESCRIPTION
**Pull request by mslabek**
## Checklist before merge
- [ ] Related issue ID is placed at the beginning of PR title in \[brackets\] (can be GH issue or Nu Jira issue)
- [ ] Code is cleaned from temporary changes and commented out lines
- [ ] Parts of the code that are not easy to understand are documented in the code
- [ ] Changes are covered by automated tests
- [ ] Showcase in dev-application.conf added to demonstrate the feature
- [ ] Documentation added or updated
- [ ] Added entry in _Changelog.md_ describing the change from the perspective of a public distribution user
- [ ] Added _MigrationGuide.md_ entry in the appropriate subcategory if introducing a breaking change
- [ ] Verify that PR will be squashed during merge

### CODE REVIEW

File: components-api/src/main/scala/pl/touk/nussknacker/engine/api/namespaces/NamingStrategy.scala [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+
+object NamingStrategy {
+  private val namespaceSeparator = "_"
+}
+
+object NamingStrategyProvider {
+  private val NamespacePath = "namespace"
+
+  def apply(modelConfig: Config): NamingStrategy = {
+    val namespaceOpt = if (modelConfig.hasPath(NamespacePath)) {
```
Comment (arkadius) [inline components-api/src/main/scala/pl/touk/nussknacker/engine/api/namespaces/NamingStrategy.scala:34]: It is just `modelConfig.getAs[String](NamespacePath)` (thanks to Ficus)

File: components-api/src/main/scala/pl/touk/nussknacker/engine/api/namespaces/NamingStrategy.scala [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+    case None => Some(name)
+  }
+
+}
+
+object NamingStrategy {
+  private val namespaceSeparator = "_"
+}
+
+object NamingStrategyProvider {
```
Comment (arkadius) [inline components-api/src/main/scala/pl/touk/nussknacker/engine/api/namespaces/NamingStrategy.scala:30]: I would merge it with the companion object above.

File: engine/flink/kafka-components-utils/src/main/scala/pl/touk/nussknacker/engine/kafka/source/flink/FlinkKafkaSource.scala [hunk 3]
```diff
@@ -127,6 +128,11 @@ class FlinkKafkaSource[T](
     deserializeTestData(formatter.parseRecord(topics.head, testParametersInfo.createTestRecord(flatParams)))
   }
 
+  private def prepareConsumerGroupId(nodeContext: FlinkCustomNodeContext): String = {
+    val baseName = overriddenConsumerGroup.getOrElse(ConsumerGroupDeterminer(kafkaConfig).consumerGroup(nodeContext))
+    namingStrategy.prepareName(baseName)
```
Comment (arkadius) [inline engine/flink/kafka-components-utils/src/main/scala/pl/touk/nussknacker/engine/kafka/source/flink/FlinkKafkaSource.scala:133]: We have to add a feature flag that allow smooth migration. I would pick the "opt-out" approach in this case. I mean: we add `KafkaConfig.useNamingStrategyInConsumerGroups = true`, we ensure that before the deployment of the new version, we set it to `false` in the config file. After we do the migration, we can remove it from the config and from the code as well.

File: engine/lite/kafka/runtime/src/main/scala/pl/touk/nussknacker/engine/lite/kafka/KafkaSingleScenarioTaskRun.scala [hunk 4]
```diff
@@ -48,12 +49,13 @@ class KafkaSingleScenarioTaskRun(
...
     interpreter: ScenarioInterpreterWithLifecycle[Future, Input, Output],
-    sourceMetrics: SourceMetrics
+    sourceMetrics: SourceMetrics,
+    namingStrategy: NamingStrategy
 )(implicit ec: ExecutionContext)
     extends Task
     with LazyLogging {
 
-  private val groupId = metaData.name.value
+  private val groupId = namingStrategy.prepareName(metaData.name.value)
```
Comment (arkadius) [inline engine/lite/kafka/runtime/src/main/scala/pl/touk/nussknacker/engine/lite/kafka/KafkaSingleScenarioTaskRun.scala:58]: The same as in flink's case

File: docs/installation_configuration_guide/model/Lite.md [hunk 5]
```diff
@@ -14,7 +14,7 @@ sidebar_label: "Lite"
...
 | shutdownTimeout          | Low        | duration | 10s                                           | How long to wait for graceful shutdown                                                                                                               |
 | interpreterTimeout       | Low        | duration | 10s                                           | Timeout of invocation of scenario (including enrichers) for events consumed in one poll                                                              |
-| publishTimeout           | Low        | duration | 10s                                           | Timeout on producing resulting event to Kafka                                                                                                        |
+| publishTimeout           | Low        | duration | 5s                                            | Timeout on producing resulting event to Kafka                                                                                                        |
```
Comment (lciolecki) [inline docs/installation_configuration_guide/model/Lite.md:17]: Yo didn't mention about change default behavior?

Comment (mslabek) [reply docs/installation_configuration_guide/model/Lite.md:17]: I should've added a comment for the reviewer, sorry for the confusion. I didn't change the behaviour, just was looking at `KafkaInterpreterConfig` and saw that the publishTimeout default arg didn't match with the docs. I didn't see an overwrite, so I assumed the docs are incorrect.

File: components-api/src/main/scala/pl/touk/nussknacker/engine/api/namespaces/NamingStrategy.scala (hunk 1 above)
Comment (mslabek) [reply components-api/src/main/scala/pl/touk/nussknacker/engine/api/namespaces/NamingStrategy.scala]: thanks, done :+1:

File: components-api/src/main/scala/pl/touk/nussknacker/engine/api/namespaces/NamingStrategy.scala (hunk 2 above)
Comment (mslabek) [reply components-api/src/main/scala/pl/touk/nussknacker/engine/api/namespaces/NamingStrategy.scala]: done :+1:

File: engine/flink/kafka-components-utils/src/main/scala/pl/touk/nussknacker/engine/kafka/source/flink/FlinkKafkaSource.scala (hunk 3 above)
Comment (mslabek) [reply engine/flink/kafka-components-utils/src/main/scala/pl/touk/nussknacker/engine/kafka/source/flink/FlinkKafkaSource.scala]: done :+1: I've added an entry in kafka configuration docs and in migration guide about the feature flag. I've named it `useNamingStrategyForConsumerGroupId`

File: engine/lite/kafka/runtime/src/main/scala/pl/touk/nussknacker/engine/lite/kafka/KafkaSingleScenarioTaskRun.scala (hunk 4 above)
Comment (mslabek) [reply engine/lite/kafka/runtime/src/main/scala/pl/touk/nussknacker/engine/lite/kafka/KafkaSingleScenarioTaskRun.scala]: done 👍

**Review by arkadius — APPROVED** [review]
Good job

**Pull request merged**